packages/babel-core/src/transformation/util/clone-deep.ts

```diff
@@ -1,25 +1,39 @@
-//https://github.com/babel/babel/pull/14583#discussion_r882828856
-function deepClone(value: any, cache: Map<any, any>): any {
+// https://github.com/babel/babel/pull/14583#discussion_r882828856
+function deepClone(
+  value: any,
+  cache: Map<any, any>,
+  allowCircle: boolean,
```

Comment:
The allowCircle is false but the object actually contains a cycle, this function is now going to cause an infinite loop, right? Can we throw in that case instead? We'd only need to keep track of the current "stack", rather than of all nodes in the map.